pandas/tests/frame/test_operators.py

```diff
@@ -218,6 +218,20 @@ def test_logical_with_nas(self):
         expected = Series([True, True])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "left, right, expected",
+        [
+            ([True, False, np.nan], [True, False, True], [True, False, False]),
+            ([True, False, True], [True, False, np.nan], [True, False, True]),
+        ],
+    )
+    def test_logical_operators_nans(self, left, right, expected):
```

Comment:
can you add tests for and as well

Reply:
Thanks for the feedback, @jreback - done.